lib/rules/layer-name-pattern/__tests__/index.mjs

```diff
@@ -0,0 +1,149 @@
+import rule from '../index.mjs';
+
+const { ruleName, messages } = rule;
+
+testRule({
+	ruleName,
+	config: /^[a-z][a-z0-9-]*$/,
+	accept: [
+		{ code: '@layer foo;' },
+		{ code: '@layer {}' },
+		{ code: '@layer foo {}' },
+		{ code: '@layer foo-bar {}' },
+		{ code: '@layer foo, bar {}' },
+		{ code: '@layer foo.bar {}' },
+		{ code: '@layer foo.bar.baz {}' },
+		{ code: '@import "foo.css" layer(foo)' },
+		{ code: '@import "foo.css" layer(foo, bar)' },
+		{ code: '@import "foo.css" layer(foo.bar)' },
+		{ code: '@import "foo.css" layer(foo.bar.baz)' },
+	],
+	reject: [
+		{
+			code: '@layer Foo {}',
+			message: messages.expected('Foo', '/^[a-z][a-z0-9-]*$/'),
+			line: 1,
+			column: 8,
+			endLine: 1,
+			endColumn: 11,
+		},
+		{
+			code: '@layer foo_bar {}',
+			message: messages.expected('foo_bar', '/^[a-z][a-z0-9-]*$/'),
+			line: 1,
+			column: 8,
+			endLine: 1,
+			endColumn: 15,
+		},
+		{
+			code: '@layer foo.Bar {}',
+			message: messages.expected('Bar', '/^[a-z][a-z0-9-]*$/'),
```

Comment:
[question] @jeddy3 says below in #8431:

> Do we want to check the whole layer name or split the period-separated list of `<ident>`s first?:
> ```css
> @layer framework.theme {}
> ```
> My hunch is that we should check the whole name

But this case seems to be checking not `foo.Bar` (the whole name) but `Bar`. Is this expected?

Reply:
Yes, let's check the whole name. 

If we added `layer-name-ident-allowed-list`, we should split there.

Reply:
Thanks for the clarification. I've applied it.
[period-separated lists are not supported](https://github.com/stylelint/stylelint/pull/8474/commits/25188aa7f4ea3125f370f730111ce63fb8c22f66)